src/assessments/seo/textImagesAssessment.js

```diff
@@ -102,16 +148,55 @@ class TextImagesAssessment extends Assessment {
 			};
 		}
 
-		// Has alt-tag, but no keyword is set
-		if ( this.altProperties.withAlt > 0 ) {
+		// Image count ≥5, has alt-tags with too few keywords.
+		if ( this.hasTooFewMatches() ) {
 			return {
-				score: this._config.scores.withAlt,
-				resultText: i18n.dgettext(
+				score: this._config.scores.withAltTooFewKeywordMatches,
+				resultText: i18n.sprintf( i18n.dgettext(
 					"js-text-analysis",
-					"The images on this page contain alt attributes."
+					"Only in %1$d out of %2$d images on this page contain alt attributes with the focus keyword. " +
+					"Where appropriate, try to include the focus keyword in more alt attributes."
+					),
+					this.altProperties.withAltKeyword,
+					this.imageCount
 				),
 			};
 		}
+
+		/*
+		 * This check needs to be made before the check for too many matches because of the special rule for
+		 * exactly 5 matches.
+		 */
+		if ( this.hasGoodNumberOfMatches() ) {
+			return {
+				score: this._config.scores.withAltGoodNumberOfKeywordMatches,
+				resultText: i18n.sprintf( i18n.dngettext(
+					"js-text-analysis",
+					"The image on this page contains an alt attribute with the focus keyword.",
+					"%1$d out of %2$d images on this page contain alt attributes with the focus keyword.",
+					this.imageCount
+					),
+					this.altProperties.withAltKeyword,
+					this.imageCount
+				),
+			};
+		}
+
+		if ( this.hasTooManyMatches() ) {
+			return {
+				score: this._config.scores.withAltTooFewKeywordMatches,
+				resultText: i18n.sprintf( i18n.dgettext(
+					"js-text-analysis",
+					"%1$d out of %2$d images on this page contain alt attributes with the focus keyword. " +
+					"That's a bit much. Only include the focus keyword when it really fits the image."
```

Comment:
Please add comments for translators

Reply:
As discussed, please change the translator comment for %2$d from "the number of images that don't contain an alt attribute with the keyword" to "the total number of images".